### PR TITLE
Storyscript Rebranding

### PR DESCRIPTION
**Pull request by kenneth-reitz**
See #139

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -36,29 +37,29 @@
     'emoji==0.5.0',
     'raven==6.9.0',
     'requests==2.20.0',
-    'storyscript==0.16.1',
+    'storyscript>=0.18.0',
```
Comment (judepereira) [inline setup.py]: Not a good idea, because unless I test some stuff in the latest release, it's not wise to put it in the CLI.

File: setup.py [hunk 2]
```diff
@@ -36,29 +37,29 @@
...
-      download_url='https://github.com/asyncy/cli/archive/master.zip',
+      download_url='https://github.com/storyscript/cli/archive/master.zip',
       keywords=' '.join(keywords),
-      author='Asyncy',
-      author_email='hello@asyncy.com',
-      url='http://docs.asyncy.com/cli',
-      license='MIT',
+      author='Storyscript',
+      author_email='hello@storyscript.io',
+      url='http://docs.storyscript.com/cli',
```
Comment (judepereira) [inline setup.py]: Our domain is storyscript.io, not storyscript.com

File: story/awesome.py [hunk 3]
```diff
@@ -28,7 +28,7 @@
     'bhaskara', 'blackburn', 'blackwell', 'bohr', 'booth', 'borg', 'bose',
     'boyd', 'brahmagupta', 'brattain', 'brown', 'buck', 'burnell', 'cannon',
     'carson', 'cartwright', 'chandrasekhar', 'chaplygin', 'chatelet',
-    'chatterjee', 'chebyshev', 'cocks', 'cohen', 'chaum', 'clarke',
+    'chatterjee', 'chebyshev', 'cohen', 'chaum', 'clarke',
```
Comment (judepereira) [inline story/awesome.py]: Haha, nice.

File: story/cli.py [hunk 4]
```diff
@@ -96,18 +97,19 @@ def track(event_name, extra: dict = None):
 
     extra['CLI version'] = version
     _make_tracking_http_request(
-        'https://stories.asyncyapp.com/track/event', {
+        'https://stories.storyscript.io/track/event', {
```
Comment (judepereira) [inline story/cli.py]: Our app domain is `storyscriptapp.com`, so this URL will be `stories.storyscriptapp.com`. `Stories` is just a Storyscript app which we have.

File: story/cli.py [hunk 5]
```diff
@@ -166,40 +170,41 @@ def initiate_login():
...
+            'Hello! Story Cloud is in private beta at this time.')
         click.echo(
             'We\'ve added you to our beta testers queue, '
             'and you should hear from us\nshortly via email'
             ' (which is linked to your GitHub account).'
         )
         sys.exit(1)
 
-    write(res.text, f'{home}/.config')
+    settings_set(r.text, f'{home}/.config')
```
Comment (judepereira) [inline story/cli.py]: Can we change .config to just config? It's already under a directory named .storyscript.

File: setup.py (hunk 2 above)
Comment (kenneth-reitz) [reply setup.py]: nice catch

File: story/cli.py (hunk 4 above)
Comment (kenneth-reitz) [reply story/cli.py]: ah, gotcha!

File: story/cli.py (hunk 5 above)
Comment (kenneth-reitz) [reply story/cli.py]: sounds good to me

File: setup.py (hunk 1 above)
Comment (kenneth-reitz) [reply setup.py]: so do we want ==0.16.1?

File: README.md [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
-[![PyPI](https://img.shields.io/pypi/v/asyncy.svg?style=flat-square)](https://pypi.org/project/asyncy/)
-[![Snap Status](https://build.snapcraft.io/badge/asyncy/snapcraft.svg)](https://build.snapcraft.io/user/asyncy/snapcraft)
-[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fasyncy%2Fcli.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2Fasyncy%2Fcli?ref=badge_shield)
+[![Requires.io](https://img.shields.io/requires/github/storyscript/cli.svg?style=flat-square)](https://requires.io/github/storyscript/cli/requirements/?branch=master)
+[![CircleCI](https://img.shields.io/circleci/project/github/storyscript/cli.svg?style=flat-square)](https://circleci.com/gh/storyscript/cli)
+[![Codecov](https://img.shields.io/codecov/c/github/storyscript/cli.svg?style=flat-square)](https://codecov.io/gh/storyscript/cli)
+[![PyPI](https://img.shields.io/pypi/v/storys.svg?style=flat-square)](https://pypi.org/project/storyscript/)
+<!-- [![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fstoryscript%2Fcli.svg?type=shield)](https://app.fossa.io/projects/git%2Bgithub.com%2Fstoryscript%2Fcli?ref=badge_shield) -->
```
Comment (wilzbach) [inline README.md]: Better fully remove it if we don't want to use this shield anymore as otherwise it will just confuse people in the future.

File: README.md [hunk 7]
```diff
@@ -16,24 +15,24 @@ The goals of this project is to provide a utility for developers to interact wit
...
 
 ✨🍰✨
 
 ## Usage
 
-Call `asyncy` to get a full list of commands or continue to [read the documentation](https://docs.asyncy.com/cli).
+Call `story` to get a full list of commands or continue to [read the documentation](https://docs.storyscript.io/cli).
 
-![usage](https://user-images.githubusercontent.com/2041757/42899845-8fe6a3a4-8ac7-11e8-8545-a22f99563368.png)
+![usage](https://github.com/storyscript/cli/blob/e7c2dc8f4b1de08163e94db16219a90fdb1d7d6b/ext/story-cli.png?raw=true)
```
Comment (wilzbach) [inline README.md]: Shouldn't we use markdown here, s.t. it's easier to update this in the future?

File: README.md [hunk 8]
```diff
@@ -59,5 +58,5 @@ asyncy
 ```
 
 
-## License
-[![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fasyncy%2Fcli.svg?type=large)](https://app.fossa.io/projects/git%2Bgithub.com%2Fasyncy%2Fcli?ref=badge_large)
\ No newline at end of file
+<!-- ## License -->
+<!-- [![FOSSA Status](https://app.fossa.io/api/projects/git%2Bgithub.com%2Fstoryscript%2Fcli.svg?type=large)](https://app.fossa.io/projects/git%2Bgithub.com%2Fstoryscript%2Fcli?ref=badge_large) -->
```
Comment (wilzbach) [inline README.md]: Again. I would just remove this

File: setup.py [hunk 9]
```diff
@@ -1,9 +1,11 @@
 # -*- coding: utf-8 -*-
+import os
+import io
```
Comment (wilzbach) [inline setup.py]: Some Python people had the great idea that imports shall be sorted and this is even one of the default checks of e.g. pylint.

File: setup.py [hunk 10]
```diff
@@ -19,13 +21,14 @@
...
-    'asyncy',
+    'storyscript',
     'devops',
     'devtools',
     'microservices',
     'orchestration',
     'serverless',
-    'storyscript',
+    'APIs',
+    'asyncy',
```
Comment (wilzbach) [inline setup.py]: Maybe add 'story' too?

File: setup.py [hunk 11]
```diff
@@ -35,30 +38,39 @@
...
-          'console_scripts': ['asyncy=cli.main:cli']
-      })
+setup(
+    name='story',
+    version=version,
+    description='Storyscript Cloud CLI',
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    classifiers=classifiers,
+    download_url='https://github.com/storyscript/cli/archive/master.zip',
```
Comment (wilzbach) [inline setup.py]:
```suggestion
    download_url=f'https://github.com/storyscript/cli/archive/{version}.zip',
```

File: setup.py [hunk 12]
```diff
@@ -35,30 +38,39 @@
...
+    keywords=' '.join(keywords),
+    author='Storyscript',
+    author_email='hello@storyscript.io',
+    url='http://docs.storyscript.io/cli',
```
Comment (wilzbach) [inline setup.py]:
```suggestion
    url='https://docs.storyscript.io/cli',
```

File: story/awesome.py (hunk 3 above)
Comment (wilzbach) [reply story/awesome.py]: See also: https://github.com/storyscript/cli/issues/124

File: story/cli.py [hunk 13]
```diff
@@ -5,36 +5,35 @@
...
-        'https://007e7d135737487f97f5fe87d5d85b55@sentry.io/1206504'
-    )
+    sentry = Client('https://007e7d135737487f97f5fe87d5d85b55@sentry.io/1206504')
 else:
     enable_reporting = False
     sentry = Client()
 
 data = None
-home = os.path.expanduser('~/.asyncy')
+home = os.path.expanduser('~/.storyscript')
```
Comment (wilzbach) [inline story/cli.py]:
```suggestion
home = os.path.expanduser('~/.story')
```

`.storyscript` might be used by the actual Storyscript compiler.

File: story/cli.py [hunk 14]
```diff
@@ -234,60 +234,69 @@ def user() -> dict:
...
+        + 'This command is deprecated and will be removed'
+        + ' in a future release. Please use '
+        + click.style(f'$ {alternative}\n', fg='magenta')
+    )
 
 
 def assert_project(command, app, default_app, allow_option):
     if app is None:
-        click.echo(click.style('No Asyncy application found.', fg='red'))
+        click.echo(click.style('No Story Cloud application found.', fg='red'))
```
Comment (wilzbach) [inline story/cli.py]:
```suggestion
        click.echo(click.style('No StoryScript Cloud application found.', fg='red'))
```

For now at least ...

File: story/cli.py [hunk 15]
```diff
@@ -234,60 +234,69 @@ def user() -> dict:
...
+
         if output == b'' and process.poll() is not None:
             break
+
         if output:
             click.echo(output.strip())
 
 
 def run(cmd: str):
+
```
Comment (wilzbach) [inline story/cli.py]: noise?

File: story/cli.py [hunk 16]
```diff
@@ -234,60 +234,69 @@ def user() -> dict:
...
-            sentry.user_context({
-                'id': data['id'],
-                'email': data['email']
-            })
+            sentry.user_context({'id': data['id'], 'email': data['email']})
 
 
 def stream(cmd: str):
     process = subprocess.Popen(cmd.split(' '), stdout=subprocess.PIPE)
+
```
Comment (wilzbach) [inline story/cli.py]: It's a good idea to split off formatting changes into different PRs to keep the focus targeted.

File: story/cli.py [hunk 17]
```diff
@@ -306,19 +315,17 @@ def run(cmd: str):
...
-             help_headers_color='yellow',
-             help_options_color='magenta')
+@click.group(cls=CLI, help_headers_color='yellow', help_options_color='magenta')
 def cli():
     """
-    Hello! Welcome to Asyncy
+    Hello! Welcome to Storyscript.
 
-    We hope you enjoy and we look forward to your feedback.
+    We hope you enjoy, & we look forward to your feedback!
```
Comment (wilzbach) [inline story/cli.py]:
```suggestion
    We hope you enjoy writing your stories.
    We are looking forward to your feedback!
```

File: story/commands/config.py [hunk 18]
```diff
@@ -94,15 +103,16 @@ def set_command(variables, app, message):
...
+        click.echo('\nSetting config and deploying new release…  ', nl=False)
+        with spinner():
             release = api.Config.set(config=config, app=app, message=message)
-        click.echo(click.style('√', fg='green'))
+        click.echo(click.style('\b' + emoji.emojize(':heavy_check_mark:'), fg='green'))
         click.echo(
-            f'Deployed new release... ' +
-            click.style(f'v{release["id"]}', bold=True, fg='magenta')
+            f'Deployed new release… '
+            + click.style(f'v{release["id"]}', bold=True, fg='magenta')
```
Comment (wilzbach) [inline story/commands/config.py]: What's the motivation for moving the `+` to the next line?
That's not a very common style.

File: story/commands/maintenance.py [hunk 19]
```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+
+import click
+import emoji
+
+from blindspin import spinner
+
+from .. import api
```
Comment (wilzbach) [inline story/commands/maintenance.py]:
```suggestion
from story import api
```

File: story/commands/maintenance.py [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+@cli.cli.group()
+def maintenance():
+    """Manage the availability of your apps."""
+    pass
+
+
+@maintenance.command()
+@options.app()
+def check(app):
+    """Displays current maintinence status."""
```
Comment (wilzbach) [inline story/commands/maintenance.py]:
```suggestion
    """Displays current maintenance status."""
```

File: story/commands/maintenance.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+    cli.user()
+    click.echo(f'Fetching maintenance mode for {app}… ', nl=False)
```
Comment (wilzbach) [inline story/commands/maintenance.py]:
```suggestion
    click.echo(f'Fetching maintenance mode for {app}… ', nl=False)
```

File: story/commands/write.py [hunk 22]
```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+import pkgutil
+
+import click
+
+import emoji
+
+from .. import cli
```
Comment (wilzbach) [inline story/commands/write.py]:
```suggestion
from story import cli
```

File: story/version.py [hunk 23]
```diff
@@ -1,2 +1,2 @@
 # -*- coding: utf-8 -*-
-version = '0.9.0'
+version = '0.9.1'
```
Comment (wilzbach) [inline story/version.py]:
```suggestion
version = '0.10.0'
```

This won't be a patch release ;-)

File: README.md (hunk 6 above)
Comment (kenneth-reitz) [reply README.md]: it just needs to be fixed.

File: README.md (hunk 7 above)
Comment (kenneth-reitz) [reply README.md]: we are using markdown?

File: setup.py (hunk 9 above)
Comment (kenneth-reitz) [reply setup.py]: not important to me, but def preferred.

File: setup.py (hunk 10 above)
Comment (kenneth-reitz) [reply setup.py]: i think that's kind of redundant, given that it's already the name of the library, it'll come up first in search :)

**Comment by kenneth-reitz** [comment]
I generally keep PRs very small, but this is just a big over-haul. Had to be done.

File: README.md (hunk 7 above)
Comment (wilzbach) [reply README.md]: You are editing a Markdown document ;-)
My point was markdown instead of the terminal help screenshot.

File: story/commands/config.py (hunk 18 above)
Comment (kenneth-reitz) [reply story/commands/config.py]: it's the way `black` formats it, which is the most popular style. I don't format code. I use black.

File: README.md (hunk 7 above)
Comment (kenneth-reitz) [reply README.md]: Ah, potentially. Let's revisit that later.

File: README.md (hunk 6 above)
Comment (kenneth-reitz) [reply README.md]: resolving this for now, we'll revisit very soon.
